crates/miden-objects/src/account/component/mod.rs

```diff
@@ -16,35 +16,37 @@ use crate::{AccountError, Word};
 // IMPLEMENTATIONS
 // ================================================================================================
 
-impl TryFrom<Package> for AccountComponentTemplate {
+impl TryFrom<&Package> for AccountComponentMetadata {
     type Error = AccountError;
 
-    fn try_from(package: Package) -> Result<Self, Self::Error> {
-        let library = package.unwrap_library().as_ref().clone();
-
-        // Look for account component metadata in sections
-        let metadata = package
+    fn try_from(package: &Package) -> Result<Self, Self::Error> {
+        package
             .sections
             .iter()
             .find_map(|section| {
                 (section.id == SectionId::ACCOUNT_COMPONENT_METADATA).then(|| {
-                        AccountComponentMetadata::read_from_bytes(&section.data)
-                            .map_err(|err| {
-                                AccountError::other_with_source(
-                                    "failed to deserialize account component metadata",
-                                    err,
-                                )
-                            })
+                    AccountComponentMetadata::read_from_bytes(&section.data).map_err(|err| {
+                        AccountError::other_with_source(
+                            "failed to deserialize account component metadata",
+                            err,
+                        )
+                    })
                 })
             })
             .transpose()?
             .ok_or_else(|| {
                 AccountError::other(
                     "package does not contain account component metadata section - packages without explicit metadata may be intended for other purposes (e.g., note scripts, transaction scripts)",
                 )
-            })?;
+            })
+    }
+}
+
+impl TryFrom<Package> for AccountComponentMetadata {
```

Comment:
Nit: I prefer not having these implementations from owned objects if the actual implementation requires just a reference, because it could easily lead to unnecessarily cloning the package.